### ISSUE TITLE
Upgrading cluster 1.20.0 <-> 1.19.0 Loop

### ISSUE THREAD

**Issue by mdegouw** [cloudnative-pg/cloudnative-pg#1993]
I deployed a basic cluster trough and the cluster in constantly upgrading itself between cloudnative-pg:1.19.0 and cloudnative-pg:1.20.0

**cloudnative-pg helm**
```
---
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: cloudnative-pg
  namespace: default
spec:
  interval: 15m
  chart:
    spec:
      chart: cloudnative-pg
      version: 0.18.0
      sourceRef:
        kind: HelmRepository
        name: cloudnative-pg
        namespace: flux-system
  maxHistory: 2
  install:
    createNamespace: true
    remediation:
      retries: 3
  upgrade:
    cleanupOnFail: true
    remediation:
      retries: 3
  uninstall:
    keepHistory: false
  values:
    crds:
      create: true
    config:
      data:
        INHERITED_ANNOTATIONS: kyverno.io/ignore
```

**Cluster Yaml**
```
---
apiVersion: postgresql.cnpg.io/v1
kind: Cluster
metadata:
  name: postgres
  namespace: default
  annotations:
    kyverno.io/ignore: "true"
spec:
  instances: 3
  imageName: ghcr.io/cloudnative-pg/postgresql:15.2
  inheritedMetadata:
    labels:
      kube-image-keeper.enix.io/image-caching-policy: ignore
  primaryUpdateStrategy: unsupervised
  storage:
    size: 20Gi
    storageClass: freenas-iscsi-csi
  superuserSecret:
    name: cloudnative-pg-secret
  # bootstrap:
  #   recovery:
  #     source: &previous-cluster postgres-v1
  postgresql:
    parameters:
      max_connections: "600"
      shared_buffers: 512MB
  monitoring:
    enablePodMonitor: true
  backup:
    retentionPolicy: 30d
    barmanObjectStore:
      wal:
        compression: bzip2
        maxParallel: 8
      destinationPath: s3://mdegouw-testing03/
      endpointURL: https://s3.us-west-004.backblazeb2.com
      serverName: postgres-v1
      s3Credentials:
        region:
          name: cloudnative-pg-secret
          key: aws-region
        accessKeyId:
          name: cloudnative-pg-secret
          key: aws-access-key-id
        secretAccessKey:
          name: cloudnative-pg-secret
          key: aws-secret-access-key
  # Note: externalClusters is needed when recovering from an existing cnpg cluster
  # externalClusters:
  #   - name: *previous-cluster
  #     barmanObjectStore:
  #       wal:
  #         compression: bzip2
  #         maxParallel: 8
  #       destinationPath: s3://mdegouw-testing03/
  #       endpointURL: https://s3.us-west-004.backblazeb2.com
  #       s3Credentials:
  #         region:
  #           name: cloudnative-pg-secret
  #           key: aws-region
  #         accessKeyId:
  #           name: cloudnative-pg-secret
  #           key: aws-access-key-id
  #         secretAccessKey:
  #           name: cloudnative-pg-secret
  #           key: aws-secret-access-key

```

**Here you can see thats its upgrading node postgres-3 from 1.19.0 to 1.20.0**
```
Cluster Summary
Name:               postgres
Namespace:          default
System ID:          7227915366007984160
PostgreSQL Image:   ghcr.io/cloudnative-pg/postgresql:15.2
Primary instance:   postgres-1
Status:             Upgrading cluster Restarting instance postgres-3, because: the instance is using an old init container image: ghcr.io/cloudnative-pg/cloudnative-pg:1.19.0 -> ghcr.io/cloudnative-pg/cloudnative-pg:1.20.0
Instances:          3
Ready instances:    2
Current Write LSN:  0/E0630B0 (Timeline: 1 - WAL File: 00000001000000000000000E)
```

**And a couple of minutes later it's "upgrading" the postgres-3 from 1.20.0 to 1.19.0**
```
Cluster Summary
Name:               postgres
Namespace:          default
System ID:          7227915366007984160
PostgreSQL Image:   ghcr.io/cloudnative-pg/postgresql:15.2
Primary instance:   postgres-1
Status:             Upgrading cluster Restarting instance postgres-3, because: the instance is using an old init container image: ghcr.io/cloudnative-pg/cloudnative-pg:1.20.0 -> ghcr.io/cloudnative-pg/cloudnative-pg:1.19.0
Instances:          3
Ready instances:    2
Current Write LSN:  0/F01A020 (Timeline: 1 - WAL File: 00000001000000000000000F)
```

Did i miss something?

**Comment by sxd** [comment]
Hi @mdegouw 

Can you confirm if this still happening ? 

Regards,

**Comment by mdegouw** [comment]
Hi @sxd 

Yes it is still happening.

It updates/downgrades the instances one by one constantly.

**Comment by gbartolini** [comment]
We have made several improvements and changes in the meantime. Can you please confirm it is still happening @mdegouw? thanks.

**Comment by sxd** [comment]
Probably this has already been fixed with the latest release 1.22, please open a new issue if this happens again
